### PR TITLE
v3. Add options

### PR DESCRIPTION
**Pull request by shyam-habarakada**
Add the ability to optionally hide,

* likes
* event interests (actual event attendance is not filtered out)
* shares

I’ve also added an experimental feature to filter posts based on
keyword matching.

The new options page looks like,

image

Hidden content is logged to the console instead of the previous blue rectangle because with more and more unwanted content hidden, these rectangle start to dominate your feed space. The console messages will appear like,

image

### CODE REVIEW

**Comment by shyam-habarakada** [comment]
I'm getting some false positives on the case insensitive comparisons. Testing...

**Comment by ovidiuch** [comment]
@shyam-habarakada is this ready? I think I saw a message from you somewhere (Twitter?) but I can't find it anymore. 

Btw, I made you a collaborator a while ago but maybe you missed the email notification
image

**Comment by shyam-habarakada** [comment]
@skidding yes, I missed the notification it seems. Thanks.

I think this is ready. I just noticed the blocking of shared content is also blocking _shared memories_ which I think is something we should not hide. I can update the filter logic for that. Will update the PR and ping you.

**Comment by shyam-habarakada** [comment]
@skidding I updated the README.

How do you publish this? I assume via the Chrome Developer Dashboard? I'm wondering if auto-update is enabled https://developer.chrome.com/apps/autoupdate

If yes, I think this should be good to merge.

File: README.md [hunk 1]
```diff
@@ -1,6 +1,38 @@
...
-[@skidding](https://twitter.com/skidding)
\ No newline at end of file
+Once installed, sponsored and suggested posts (typically ads or promoted pages) will be removed from your news feed.
+
+Options
+-------
+
+The extension provides options to also hide content that fall in the categories listed below.
+
+The thinking behind these additional features is that we want to enable you to take even more more control of what you see in your feed. We take the point of view that this means seeing actual conversations your friends are having on Facebook. Posts people are making where they write something, photos being posted, comments fall in the category of _actual conversations_. The notifications feature in facebook is a great way to find out when someone likes content _you_ or tags you in something they shared. These are part of actual conversation mechnisms we want to continue to use, and the options below do not interfere with notifications.
```
Comment (ovidiuch) [inline README.md]: *mechanisms

File: options.html [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      font-size: 130%;
+    }
+    .statusWrapper {
+      height: 50px;
+    }
+  </style>
+</head>
+<body>
+<h1>Hide Facebook Sponsored Posts ++</h1>
+<p>In addition to hiding ads and sponsored posts, you can customize this extension to hide links shared by people or pages you follow as well as likes activities. If you want even more customization, you can enter one or more keywords separated by commas and we will attempt to hide any posts that match those keywords. Example keywrods can be like "trump,hate". Keyword matching is case insensitive.
```
Comment (ovidiuch) [inline options.html]: s/keywrods/keywords

File: README.md [hunk 3]
```diff
@@ -1,6 +1,38 @@
...
+
+Improvements
+------------
+
+A few things we want to improve that are outstanding at the moment,
+
+* Don't hide shared memories. These are fun and is one reason we want to keep using facebook.
+
+
+[@skidding](https://twitter.com/skidding)
```
Comment (ovidiuch) [inline README.md:38]: I think you should add your name as well @shyam-habarakada

**Review by ovidiuch — COMMENTED** [review]
Cool stuff!

Just a suggestion, I recommend adding Prettier to this codebase as it improved my developer experience in every since project I worked on recently. Regardless on your decision on this, let me know after you fix the typos and I'll merge this.

I use Chrome Developer Dashboard so auto-updates should be on. I'll investigate if I can give you rights to publish there.

**Comment by ovidiuch** [comment]
> I'll investigate if I can give you rights to publish there.

Doesn't seem like it. If you know of a way that allows you to publish updates without me let me know, otherwise I'll upload the zip when you need it.

**Comment by shyam-habarakada** [comment]
@skidding addressed the last comments. I looked at https://prettier.io/docs/en/install.html but it's not immediately obvious to me how this fits into my workflow (I'm in old school vim :-) If you have pointers (or when I get to looking a bit closer at it) I can add that to the repo.

**Comment by ovidiuch** [comment]
Looks good to me! You can merge this PR yourself since you have collaborator rights now.

Regarding Prettier, there seems to be a [Vim Setup](https://prettier.io/docs/en/vim.html). Regardless if that works for you, setting up a [Pre-commit Hook](https://prettier.io/docs/en/precommit.html) is a standard approach. Feel free to try this any time in the future, no need to stall this PR.

**Comment by shyam-habarakada** [comment]
Merged. Can you publish? Great if we can update the screenshots in the web store as well. Let me know if I can help.

**Comment by ovidiuch** [comment]
@shyam-habarakada Send the screenshots my way and I'll update everything.

**Pull request merged**